src/software/ai/hl/stp/tactic/tactic_world_params_update_visitor_test.cpp

```diff
@@ -0,0 +1,236 @@
+#include "software/ai/hl/stp/tactic/tactic_world_params_update_visitor.h"
+
+#include <gtest/gtest.h>
+
+#include "software/test_util/test_util.h"
+
+TEST(TacticUpdateVisitorTest, update_cherry_pick_tactic)
+{
+    World initial_world     = ::Test::TestUtil::createBlankTestingWorld();
+    Rectangle target_region = Rectangle(Point(1, 1), Point(0, 0));
+    World update_world      = ::Test::TestUtil::createBlankTestingWorld();
+    update_world            = ::Test::TestUtil::setBallPosition(update_world, Point(1, 0),
+                                                     Timestamp::fromSeconds(0));
+    update_world.updateFieldGeometry(
+        Field(10.0, 10.0, 10.0, 10.0, 10.0, 10.0, 10.0, Timestamp::fromSeconds(0)));
+    CherryPickTactic tactic = CherryPickTactic(initial_world, target_region);
+    EXPECT_EQ(tactic.getWorld().field(), initial_world.field());
+    EXPECT_EQ(tactic.getWorld().enemyTeam(), initial_world.enemyTeam());
+    EXPECT_EQ(tactic.getWorld().friendlyTeam(), initial_world.friendlyTeam());
+    EXPECT_EQ(tactic.getWorld().ball(), initial_world.ball());
+    TacticWorldParamsUpdateVisitor visitor = TacticWorldParamsUpdateVisitor(update_world);
+    tactic.accept(visitor);
+    EXPECT_EQ(tactic.getWorld().field(), update_world.field());
+    EXPECT_EQ(tactic.getWorld().enemyTeam(), update_world.enemyTeam());
+    EXPECT_EQ(tactic.getWorld().friendlyTeam(), update_world.friendlyTeam());
+    EXPECT_EQ(tactic.getWorld().ball(), update_world.ball());
```

Comment:
Are we just checking that the `World` is equal here? If so, we should either:
- implement an equality operator for `World`
- open a ticket to implement an equality operator for `World`, and leave a comment like `TODO (Issue #1234): use World equality operator when it's complete` here

Same goes for cases where we're checking world equality below.

Reply:
added equality operator in world

Reply:
Great! Could we make sure to add tests for all the new equality operators (I think just `World` and `GameState`)?

Reply:
added